### DISCUSSION TITLE
Dark mode support for Toolbars such as Quick Launch?

### DISCUSSION THREAD

**Discussion by chip33** [valinet/ExplorerPatcher#552] [Ideas]
I just wondered (and highly suspect) that this would be an example of a [Small matter of programming](https://en.wikipedia.org/wiki/Small_matter_of_programming).
It would certainly be a nice feature that MS never implemented.

**Comment by valinet** [comment]
I don’t even understand what do you mean. What’s the deal with Quick Launch and the dark theme?

**Reply by chip33** [reply]
Making an option or otherwise supporting for the toolbar popup lists to better match a dark mode windows theme. They are always dark text on a light background.

![Screenshot 2021-12-12 151052](https://user-images.githubusercontent.com/48499740/145718306-86c190fe-d898-44a9-b417-2c9229d24a43.png)

**Reply by valinet** [reply]
Yeah, idk, maybe doable... everything is doable after all... the window that owns those menus basically has to have its theme set to `Explorer` and the undocumented and weirdly working `SetPreferredAppMode` and `AllowDarkModeForWindow` called on it and maybe then it has a chance of working. It seems those things appear to skin to dark the menus in File Explorer when owner drawing is disabled, so maybe there is a possibility of making it work here as well. I don't know exactly where that functionality is, let alone having tested my hypothesis above.

**Reply by valinet** [reply]
Yeah, looked on it, the menu is not created using `user32!TrackPopupMenu` or `user32!NtUserTrackPopupMenuEx`, it uses something else, so it would not automatically benefit from dark theme support for context menus in the `Explorer` theme, so it basically means implementing that from scratch. So yeah, far from being [a small matter of programming](https://en.wikipedia.org/wiki/Small_matter_of_programming), or at least that's how it looks to me at the moment.

**Comment by chip33** [comment]
Found a great open source alternative and huge improvement to Quick Launch. [SystemTrayMenu](https://github.com/Hofknecht/SystemTrayMenu)
![SystemTrayMenu_1](https://user-images.githubusercontent.com/48499740/150677109-4e6715aa-5ad4-4ca5-8f3a-67797e31fde9.png)

**Reply by valinet** [reply]
Interesting. You know, a Start menu replacement can also basically do the same thing: most offer a list where you can pin items
